### DISCUSSION TITLE
Resolving .js paths in TS project

### DISCUSSION THREAD

**Discussion by fregante** [web-infra-dev/rsbuild#3529] [Q&A] [answered]
```ts
// index.ts
import './file.js'; // <-- "wrong" extension
```
```ts
// file.ts
console.log('hi')
```

It seems that this basic example works in a new `create-rsbuild` project, but fails in my existing project. I tried copying the entire tsconfig to no avail, I get this:

```
File: ~/source/refined-github.ts:1:1
  × Module not found: Can't resolve './features/linkify-line-numbers.js' in '~/source'
     ╭─[215:7]
 213 │ import './features/click-outside-modal.js';
 214 │ import './features/comment-excess.js';
 215 │ import './features/linkify-line-numbers.js';
     ·        ────────────────────────────────────
     ╰────
```

Why would this be happening? The paths are correct, in fact it works if I replace .js with .tsx, but this shouldn't be necessary.

See https://github.com/refined-github/refined-github/pull/7817

**Comment by chenjiahan** [comment, marked as answer]
Rspack provides [resolve.extensionAlias](https://rspack.dev/config/resolve#resolveextensionalias) as a solution to TS extension alias.

Example:

```js
export default defineConfig({
  tools: {
    rspack: {
      resolve: {
        extensionAlias: {
          '.js': ['.ts', '.js'],
          '.jsx': ['.tsx', '.jsx'],
        },
      },
    },
  },
});
```

We are considering adding this configuration by default in Rsbuild, until then, you can add it manually to solve this problem.
